vioscsi/helper.c

```diff
@@ -56,21 +56,25 @@ SendSRB(
     UCHAR               ScsiStatus = SCSISTAT_GOOD;
     ULONG MessageID;
     int res = 0;
+    BOOLEAN             notify = FALSE;
     PREQUEST_LIST       element;
     ULONG               index;
-ENTER_FN_SRB();
 
-    if (!Srb)
-        return;
+ENTER_FN_SRB();
 
+    if (!Srb) {
+EXIT_FN_SRB();
```

Comment:
The question here I think is: does it stay or does it go...? Another call slows the driver down.
Maybe this is technically a macro, so normally they wouldn't be indented methinks.
@vrozenfe can probably answer this.

Reply:
It's probably worth mentioning that we can probably add some conditional blocks like `#ifndef TRACE_VQLOCKS` or `#ifndef TRACE_SRBS` to wrap these. Then indenting might be less of an issue.

Reply:
The other option here, if everyone is ok with using it would be to use `goto label_at_end_of_function;` and then drop `label_at_end_of_function:` just before the WPP macro call. I'm fine with that. It's really what `goto` in C is supposed to be used for and cannot be used to jump out of the function like in other langs. I see it in [virtio] in some places. Many projects frown on using `goto` though, so...

Reply:
... `goto` is also faster - which I'm guessing we all like...

Reply:
Alternatively, we can just drop an appropriately indented `RhelDbgPrint();` and leave it at that.
If we use say `RhelDbgPrint(TRACE_LEVEL_RESERVED8, " After KickPrep Notfy State  : %d \n", notify);` that will also work with WPP provided `TRACE_LEVEL_RESERVED8` is defined in `WPP_DEFINE_CONTROL_GUID()`. You can then just toggle that flag to enable or disable the trace.

Any of these work for me. What I think is most important is that there is working tracing...